File: Solution/【001】两数之和/两数之和.scala
import scala.collection.mutable

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        if(nums == null || nums.length == 0){
            return 0;
        }
        val hashmap: mutable.HashMap[Int, Int] = mutable.HashMap()
        for (i <- 0 until nums.length){
            val div = target - nums(i)
            if(hashmap.contains(div)){
                return Array(hashmap(div), i)
            }
            else{
                hashmap(nums(i)) = i
            }
        }
        return Array(-1, -1)
    }
}

File: Solution/【001】两数之和/两数之和.cpp


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> my_map;
        for(int i = 0; i <= nums.size(); i++){
            int div = target - nums[i];
            if(my_map.find(div) != my_map.end()){
                result.push_back(my_map[div]);
                result.push_back(i);
                return result;
            }
            my_map[nums[i]] = i;
        }
        return result;
    }
};